pkg/txhandler/simple/simple_transaction_handler.go

```diff
@@ -0,0 +1,473 @@
+	var info simplePolicyInfo
+	infoBytes := []byte(mtx.PolicyInfo.String())
+	if len(infoBytes) > 0 {
+		err := json.Unmarshal(infoBytes, &info)
+		if err != nil {
+			log.L(ctx).Warnf("Failed to parse existing info `%s`: %s", infoBytes, err)
+		}
+	}
+	update, reason, err = fn(&info)
+	if update != UpdateNo {
+		infoBytes, _ = json.Marshal(&info)
+		mtx.PolicyInfo = fftypes.JSONAnyPtrBytes(infoBytes)
+	}
+	return update, reason, err
+}
+
+func (sth *simpleTransactionHandler) Init(ctx context.Context, toolkit *txhandler.Toolkit) {
+	sth.toolkit = toolkit
+}
+
+func (sth *simpleTransactionHandler) Start(ctx context.Context) (done <-chan struct{}, err error) {
+	if sth.ctx == nil { // only start once
+		sth.ctx = ctx // set the context for policy loop
+		sth.policyLoopDone = make(chan struct{})
+		sth.markInflightStale()
+		go sth.policyLoop()
+	}
+	return sth.policyLoopDone, nil
+}
+func (sth *simpleTransactionHandler) HandleNewTransaction(ctx context.Context, txReq *apitypes.TransactionRequest) (mtx *apitypes.ManagedTX, err error) {
+
+	// Prepare the transaction, which will mean we have a transaction that should be submittable.
+	// If we fail at this stage, we don't need to write any state as we are sure we haven't submitted
+	// anything to the blockchain itself.
+	prepared, _, err := sth.toolkit.Connector.TransactionPrepare(ctx, &ffcapi.TransactionPrepareRequest{
+		TransactionInput: txReq.TransactionInput,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return sth.createManagedTx(ctx, txReq.Headers.ID, &txReq.TransactionHeaders, prepared.Gas, prepared.TransactionData)
+}
+func (sth *simpleTransactionHandler) HandleNewContractDeployment(ctx context.Context, txReq *apitypes.ContractDeployRequest) (mtx *apitypes.ManagedTX, err error) {
+
+	// Prepare the transaction, which will mean we have a transaction that should be submittable.
+	// If we fail at this stage, we don't need to write any state as we are sure we haven't submitted
+	// anything to the blockchain itself.
+	prepared, _, err := sth.toolkit.Connector.DeployContractPrepare(ctx, &txReq.ContractDeployPrepareRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	return sth.createManagedTx(ctx, txReq.Headers.ID, &txReq.TransactionHeaders, prepared.Gas, prepared.TransactionData)
+}
+func (sth *simpleTransactionHandler) HandleCancelTransaction(ctx context.Context, txID string) (mtx *apitypes.ManagedTX, err error) {
+	res := sth.policyEngineAPIRequest(ctx, &policyEngineAPIRequest{
+		requestType: policyEngineAPIRequestTypeDelete,
+		txID:        txID,
+	})
+	return res.tx, nil
+}
+
+func (sth *simpleTransactionHandler) GetTransactionByID(ctx context.Context, txID string) (transaction *apitypes.ManagedTX, err error) {
+	tx, err := sth.toolkit.Persistence.GetTransactionByID(ctx, txID)
```

Comment:
See other comments about this code being LevelDB persistence specific, rather than TX handler specific.

There's debt here (unrelated to your changes) where the code should move to the https://github.com/hyperledger/firefly-common/pull/39 interface for filtering, and the LevelDB specific implementation of filtering in this function should be placed behind a LevelDB restricted parsing of that filter syntax. Not suggesting you take that on, but having the code in the TXHander rather than FFTM would be adding to the debt.

Reply:
agreed with the comment on debt propagation, will avoid adding this function for now